### DISCUSSION TITLE
OpenAPI Response spec

### DISCUSSION THREAD

**Discussion by perfectra1n** [TriliumNext/Trilium#3745] [Q&A]
Hi zadam,

I'm trying to set up an automated pipeline that generates a Python "wrapper" client automatically, provided the OpenAPI spec document.

When utilizing a tool such as `openapi-python-client` to generate it, it throws the following warning and does not generate a client portion for the API path:

```
WARNING parsing POST /auth/login within default.

Invalid response status code default (not a valid HTTP status code), response will be ommitted from generated client
```

It appears that the issue lies in the OpenAPI document like [here](https://github.com/zadam/trilium/blob/master/src/etapi/etapi.openapi.yaml#L43-L48). Is there a specific reason that it would be best to just leave that `default` response section in the `etapi.openapi.yaml` document, or would you be open to a pull request that breaks those down further to `400` and `404` responses?

I would also love to share it out and manage it from pypi if you think the community would be interested :)

**Comment by zadam** [comment]
Hi, `default` is part of the spec: https://swagger.io/docs/specification/describing-responses/

It's a good way to avoid unnecessary repetition.

**Comment by perfectra1n** [comment]
I could've sworn that I at lease "ctrl + f"d that page before I asked. Thanks for at least taking the time to respond to me lol, I feel pretty dumb.

If it makes you feel any better, I'm creating a Python SDK for interacting w/ Trilium here:
https://github.com/perfectra1n/pytrilium

I'm hoping to always have all valid ETAPI paths useable, and relatively easy to maintain. Let me know what you think of what is in there so far, and feel free to add it to your README if you like it.

**Reply by zadam** [reply]
I think it would be good to add to this list: https://github.com/Nriver/awesome-trilium#api-extensions

@Nriver seems to be working on a similar python API wrapper: https://github.com/Nriver/trilium-py

**Reply by perfectra1n** [reply]
Cool! Yeah `trilium-py` was missing the `/export` endpoint for notes when I was attempting to use it earlier, and it didn't expose the underlying `requests` session. So I just spun up my own real quick. I'll add it to the `awesome-trilium` list, I'm hoping to keep it completely feature complete to the current ETAPI.

**Reply by Nriver** [reply]
Thanks for the notice. I've merged the PR for [pytrilium](https://github.com/perfectra1n/pytrilium) to be added into [awesome-trilium](https://github.com/Nriver/awesome-trilium#api-extensions). I've also updated my [trilium-py](https://github.com/Nriver/trilium-py) for the `export` feature.
